### PR TITLE
Corrected documentation for batList {last, hd}

### LINKED ISSUE #754: BatList suggests first and last should raise Empty_list
But `first` is actually raising Failure and `last` Invalid_argument.

Empty_list is raised by LazyList and RefList though.
I guess we should catch Failure/Invalid_argument and re-raise Empty_list in `first` and `last`.

### PR DESCRIPTION
**Pull request by FkHina**
fix #754

### CODE REVIEW

File: src/batList.mliv [hunk 1]
```diff
@@ -73,19 +73,20 @@ val is_empty : 'a list -> bool
...
-    the list is empty (similar to [hd]). *)
-
 val hd : 'a list -> 'a
-(** Similar to [first], but @raise Failure if the list is empty. *)
+(** Returns the first element of the list, or @raise Failure if 
+    the list is empty. *)
+
+
+val first : 'a list -> 'a
+(** Alea to hd *)
```
Comment (gasche) [inline src/batList.mliv]: Alias?

**Comment by FkHina** [comment]
yes, it was a typo, i Will correct that

2018-02-25 20:14 GMT+01:00 Gabriel Scherer <notifications@github.com>:

**Comment by thizanne** [comment]
For the record, @FkHina is a student who will work on batteries as a software project. I will not merge her PR myself (at least not her first one).

We were not sure if this needs a Changes entry.

**Comment by FkHina** [comment]
Merci pour votre mail, normalement la correction a été bien faite

Cordialement

2018-02-25 20:47 GMT+01:00 Thibault Suzanne <notifications@github.com>:

**Comment by UnixJunkie** [comment]
such entries are "corrected documentation for ..." usually

**Comment by FkHina** [comment]
Ok then I will add them, thank you for the answer
Le 26 févr. 2018 10:53, "Gabriel Scherer" <notifications@github.com> a
écrit :
>
> @gasche approved this pull request.
>
> I have reviewed the change and believe this is correct.
>
> I think that it would be nice to add a Changes entry: documentation
changes are valuable contributions to the project, so it's good to credit
them.

**Pull request merged**